### DISCUSSION TITLE
Dinamically update template IDs of Loop items when importing/exporting Site Kits

### DISCUSSION THREAD

**Discussion by joaodath** [elementor/elementor#23805] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

> For the sake of conciseness, I'll be calling websites that **export** a Site Kit as _Origin Websites_ and websites that **import** a Site KIt as _Destination Websites_. 

When exporting a Site Kit, Elementor references each element inside the templates using IDs. However, when importing the Site Kit, Elementor expects the same IDs to be available on the Destination Website. If the IDs are not available, Site Kit imports will mostly fail without displaying a descriptive message about the error. The issue is even worse for Site Kits with Loop items. When IDs used by loop items are not available in the Destination Website, pages and templates that use them will fail hard. This behavior is described here: https://github.com/elementor/elementor/issues/22202 

### Describe the solution you'd like

As I mentioned in issue #22202 Elementor should be able to verify the IDs while importing the Site Kit and dinamically update them if needed. That way, there wouldn't be hard fails when importing Site Kits that use the Loop feature. 

Another way would be for the Site Kit import flow to check the pages and gather which templates they depend on, then check those templates for resources that they also depend on, repeat this until it finds the most basic element/template, import it and then update its ID on every template that requires it.

If you really think about this, having a file that points which resources are needed and where they are (similar to package.json and package-lock.json made by NPM in Node.js apps) so the Site Kit flow would make those resources available first could potentially fix this issue, instead of relying on hard-coded IDs to be available on Destination Websites.

### Describe alternatives you've considered

A workaround provided by Elementor Support was to disable the Loop feature, open the affected pages and redo the Loop, a very time consuming workaround if you're importing a Site Kit with lots of Loops. Another way I personally found was to manually import the Loop templates on the Destination Website, then get its ID and replace it on the Site Kit using a code editor like VSCode, but this is far too technical for a plugin that has "No Coding" as key selling point in its official page (https://elementor.com/products/page-builder-plugin/). 

### Additional context

Even though @nicholaszein mentioned here (https://github.com/elementor/elementor/issues/22202#issuecomment-1719893974) that this feature is not supported, and there's no ETA for when this will be available, I strongly disagree. This is should not be considered as a feature request, but as a bug fix request. Considering that both features, Loop elements and Site Kit, are supported features of Elementor, they have not been deprecated, nor they are beta features, they're both displayed as features of Elementor in the official page (https://elementor.com/features/) and Loop is a Pro feature (meaning we pay for that), it seems like a no-brainer that both features do not work together.

**Comment by sreidnoe** [comment]
I went through a full stack program. I’ve gotten a little lazy(ish) haha
with being able to use Elementor.

I literally spent oh about 50 hours in 4 days revamping my site for the
contest boo. This included custom loop items.

I know I’ll be too late for this go round bc I’ve lost 36 hours of time to
be able to complete and I had around 20 sprints left.

Anywhos, renaming id’s is something I can do and I’m willing to do that to
save my work. Is there a video or Zoom to point me in the right direction,
or do I just need to open the files for Elementor and then then the files
for my back up?

**Reply by codewizard13** [reply]
@nicholaszein , can you answer this question? Can you point us to an Elementor tutorial showing how to correctly import templates with loop items (which is basically my whole site)?
